plugins/woocommerce/client/blocks/assets/js/blocks/product-details/edit.tsx

```diff
@@ -109,24 +45,115 @@ const useIsInvalidQueryLoopContext = ( clientId: string, postType: string ) => {
 const Edit = ( { clientId, context }: ProductDetailsEditProps ) => {
 	const blockProps = useBlockProps();
 
+	const product = useSelect(
+		( select ) => {
+			if ( ! context.postId ) {
+				return null;
+			}
+			const { getProduct } = select( productsStore );
+			return getProduct( Number( context.postId ) );
+		},
+		[ context.postId ]
+	);
+
+	const isInnerBlockOfSingleProductBlock = useSelect(
+		( select ) => {
+			const blocks = select(
+				blockEditorStore
+				// @ts-expect-error - getBlockParentsByBlockName is not typed
+			).getBlockParentsByBlockName(
+				clientId,
+				'woocommerce/single-product'
+			);
+			return blocks.length > 0;
+		},
+		[ clientId ]
+	);
```

Comment:
As per [How to work effectively with the useSelect hook](https://developer.wordpress.org/news/2024/03/how-to-work-effectively-with-the-useselect-hook/#prefer-returning-objects-with-value-properties-from-the-callback) it's recommended returning object with values we need rather than direct value.

```suggestion
	const { product } = useSelect(
		( select ) => {
			if ( ! context.postId ) {
				return null;
			}
			const { getProduct } = select( productsStore );
			return { product: getProduct( Number( context.postId ) ) };
		},
		[ context.postId ]
	);

	const { isInnerBlockOfSingleProductBlock } = useSelect(
		( select ) => {
			const blocks = select(
				blockEditorStore
				// @ts-expect-error - getBlockParentsByBlockName is not typed
			).getBlockParentsByBlockName(
				clientId,
				'woocommerce/single-product'
			);
			return { isInnerBlockOfSingleProductBlock: blocks.length > 0 };
		},
		[ clientId ]
	);
```